src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -21,282 +26,459 @@ data class KmerMapData(val rangeToBitSetMap: Map<ReferenceRange, BitSet>, val km
 
 data class KeyFileData(val sampleName: String, val file1: String, val file2: String = "")
 
-private val myLogger = LogManager.getLogger("net.maizegenetics.phgv2.utils.AlignmentUtils")
-
-
-/**
- * Function to align the reads coming from a single ended Fastq or a pair of fastq files to the haplotype graph kmer index found in [kmerIndexFile] and create a readMapping file which will be written to the [outputDir].
-
- */
-fun alignReadsToHaplotypes(hvcfDir: String, kmerIndexFile:String, keyFileRecords:List<KeyFileData>, outputDir:String) {
-    //loop through all files in hvcfDir and create a list of hvcf files
-    val hvcfFiles = File(hvcfDir).walkTopDown().filter { it.isFile }.filter { it.name.endsWith("h.vcf") || it.name.endsWith("h.vcf.gz") }.map { "${it.path}/${it.name}" }.toList()
+private val myLogger = LogManager.getLogger("net.maizegenetics.phgv2.pathing.AlignmentUtils")
+
+class AlignmentUtils {
+    companion object {
+        /**
+         * Function to align the reads coming from a single ended Fastq or a pair of fastq files to the haplotype graph kmer
+         * index found in [kmerIndexFile] and create a readMapping file which will be written to the [outputDir].
+         */
+        fun alignReadsToHaplotypes(
+            graph: HaplotypeGraph,
+            kmerIndexFile: String,
+            keyFileRecords: List<KeyFileData>,
+            outputDir: String,
+            numThreads: Int = 5
+        ) {
+            val kmerIndexMap = loadKmerMaps(kmerIndexFile, graph)
+
+            for (keyFileRecord in keyFileRecords) {
+                val hapIdMapping = processReadMappingForKeyFileRecord(keyFileRecord, kmerIndexMap, graph, numThreads)
+
+                //export the read mapping to disk
+                //Use the first file name as the readmapping output name
+                val inputFile1 = File(keyFileRecord.file1)
+                val outputFileName = "${outputDir}/${inputFile1.nameWithoutExtension}_readMapping.txt"
+
+
+                exportReadMapping(
+                    outputFileName,
+                    hapIdMapping,
+                    keyFileRecord.sampleName,
+                    Pair(keyFileRecord.file1, keyFileRecord.file2)
+                )
+            }
 
-    //create a HaplotypeGraph from the list of hvcf files
-    val graph = HaplotypeGraph(hvcfFiles)
+        }
 
-    val kmerIndexMap = loadKmerMaps(kmerIndexFile, graph)
+        /**
+         * Loads a kmer hash map file to memory for use in mapping reads.
+         */
+        fun loadKmerMaps(filename: String, graph: HaplotypeGraph): KmerMapData {
+            //Load the contents of the file into
+            //rangeHapidMap: a map of refRange to an Array<String> of the haplotype ids in the ReferenceRange
+            // and the BitSet of all hapid sets
```

Comment:
consider adding a brief explanation or what the BitSet of all hapid sets is and how it is used.